### PR TITLE
Build unit tests and add option to omit

### PR DESCRIPTION
**Pull request by alanking**
Also renames --server-only option to core-only

### CODE REVIEW

File: build_and_copy_packages_to_dir.sh [hunk 1]
```diff
@@ -24,19 +24,21 @@ else
...
 make_program_config=""
 build_jobs=0
 debug_config=""
+unit_test_config="-DIRODS_UNIT_TESTS_BUILD=YES"
 
 while [ -n "$1" ]; do
     case "$1" in
-        --server-only)           server_only=1;;
+        --core-only)             core_only=1;;
         -N|--ninja)              make_program_config="-GNinja";
```
Comment (trel) [inline build_and_copy_packages_to_dir.sh:37]: unrelated, but i see only a single semicolon here...

Comment (alanking) [reply build_and_copy_packages_to_dir.sh:37]: This is intentional (bash-ism) in order to set `make_program` on the following line

Comment (alanking) [reply build_and_copy_packages_to_dir.sh:37]: Or so I would think. @d-w-moore, thoughts?

Comment (trel) [reply build_and_copy_packages_to_dir.sh:37]: everybody knows that

**Pull request merged**